Source/CarthageKit/GitHub.swift

```diff
@@ -266,6 +284,14 @@ internal struct GitHubCredentials {
 	}
 }
 
+internal func loadGitHubCredentials(accessToken: GitHubAccessToken?) -> SignalProducer<GitHubCredentials?, CarthageError> {
```

Comment:
What do you mean by adding complexity? I know that thre are more types but I think it makes the code easier to read and the intent more clear. 

Maybe I am missing something but I think that we would be putting too much responsability in the `loadGitHubAuthorization` method. 

Also we would have to change every method that receives as `GitHubCredentials` to receive a `String` with the value of the authorization header. Which I don't think is good but I don't have a strong argument just that I prefer having a `GitHubCredential` that knows how to build the value for the authorization header. Maybe we can do some refactoring in https://github.com/guidomb/Carthage/blob/github-access-token-option/Source/CarthageKit/Project.swift#L311 and put all the logic of getting the github credentials inside the `GitHub.swift` file.

What do you think?